lib/scholar/cluster/k_means.ex

```diff
@@ -0,0 +1,322 @@
+defmodule Scholar.Cluster.KMeans do
+  @moduledoc """
+  K-Means algorithm
+  """
+  import Nx.Defn
+  import Scholar.Shared
+
+  @derive {Nx.Container, containers: [:num_iterations, :clusters, :inertia, :labels]}
+  defstruct [:num_iterations, :clusters, :inertia, :labels]
+
+  @doc """
+  Fits a K-Means model for sample inputs `x`
+
+
+  ## Options
+
+    * `:num_clusters` - The number of clusters to form as well as the number of centroids to generate. Defaults to `8`.
+
+    * `:max_iterations` - Maximum number of iterations of the k-means algorithm for a single run. Defaults to `300`.
+
+    * `:num_runs` - Number of time the k-means algorithm will be run with different centroid seeds. The final results will be the best output of num_runs runs in terms of inertia. Defaults to `10`.
```

Comment:
Sorry for the nitpick but can you please trim the descriptions so the lines do not exceed 100 chars?